web/src/scripts/actions/tabActions.js

```diff
@@ -75,6 +75,51 @@ export const closeAllTabs = (containerId) => async (dispatch) => {
   dispatch({type: at.CLOSE_ALL_TABS, payload: {containerId}})
 }
 
+export const focusAdjacentTab = (containerId, iterator, options = {}) => async (dispatch, getState) => {
```

Comment:
I've been trying to keep this complex data structure shuffling in `TabUtils` for the most part, so that the logic of the action is easy to follow.